File: PG/[PG]올바른괄호.cpp
#include<string>
#include <stack>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    
    stack<int>mystack;
    
    for(auto i : s){
        if(mystack.empty() && i == ')') return false;
        else if(i == '(') mystack.push(i);
        else mystack.pop();
    }
    
    if(!mystack.empty()) answer = false;
    return answer;
}

File: PG/[PG]올바른괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        if(s.charAt(0) == ')' || s.charAt(s.length()-1) == '(') return false;

        Stack<Character> stack = new Stack<Character>();
        for(char c : s.toCharArray()){
            if(c == '(') stack.push(c);
            else{
                if(stack.isEmpty()) return false;
                stack.pop();
            }
        }
        if(!stack.isEmpty()) return false;
        return true;
    }
}